### PR TITLE
fix build issues and simplified/cleaned some code

### PR DESCRIPTION
**Pull request by porcelli**

### CODE REVIEW

**Comment by manstis** [comment]
Please don't forget the PR for ```droolsjbpm-build-bootstrap``` for the version changes.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/compiler/nio/CompilationRequest.java [hunk 1]
```diff
@@ -16,22 +16,30 @@
...
-    NIOWorkspaceCompilationInfo getInfo();
+    WorkspaceCompilationInfo getInfo();
 
     String getMavenRepo();
+
+    String[] getOriginalArgs();
+
+    Boolean getLogRequested();
+
+    Map getMap();
```
Comment (jhrcek) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/compiler/nio/CompilationRequest.java]: Let's avoid raw types if possible..

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/compiler/nio/impl/DefaultCompilationRequest.java [hunk 2]
```diff
@@ -113,4 +113,19 @@ public AFCliRequest getKieCliRequest() {
...
         return mavenRepo;
     }
+
+    @Override
+    public String[] getOriginalArgs() {
+        return originalArgs;
+    }
+
+    @Override
+    public Map getMap() {
```
Comment (jhrcek) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/compiler/nio/impl/DefaultCompilationRequest.java]: Let's avoid raw types, please.

File: kie-wb-common-services/kie-wb-common-services-backend/src/test/projects/kjar-2-all-resources/pom.xml [hunk 3]
```diff
@@ -37,7 +37,7 @@
             <plugin>
                 <groupId>org.kie</groupId>
                 <artifactId>kie-takari-plugin</artifactId>
-                <version>7.2.0-SNAPSHOT</version>
+                <version>8.0.0-SNAPSHOT</version>
```
Comment (psiroky) [inline kie-wb-common-services/kie-wb-common-services-backend/src/test/projects/kjar-2-all-resources/pom.xml]: It this really not being updated automatically during the build?

Comment (porcelli) [inline kie-wb-common-services/kie-wb-common-services-backend/src/test/projects/kjar-2-all-resources/pom.xml]: nope

Comment (psiroky) [inline kie-wb-common-services/kie-wb-common-services-backend/src/test/projects/kjar-2-all-resources/pom.xml]: That's wrong then.

**Comment by porcelli** [comment]
failing test should be fixed once get this merged: https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/484

**Comment by porcelli** [comment]
Jenkins please retest this

**Comment by manstis** [comment]
@porcelli You da' man :-)

**Pull request merged**